packages/ui-components/src/components/Inputs/FileUploadField.tsx

```diff
@@ -1,172 +1,232 @@
 /*
  * Tupaia
- * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
  */
 
-import React, { useRef, useState } from 'react';
-import { FormLabel } from '@material-ui/core';
-import styled from 'styled-components';
-import MuiFormHelperText from '@material-ui/core/FormHelperText';
-import { Button } from '../Button';
-import { FlexStart } from '../Layout';
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import { FormHelperText, useTheme } from '@material-ui/core';
+import { DropEvent, useDropzone } from 'react-dropzone';
 import { FilePicker } from '../Icons';
+import { Button } from '../Button';
 import { InputLabel } from './InputLabel';
 
-const HiddenFileInput = styled.input`
-  display: none; // Hide the input element without applying other styles - setting it to be small and position absolute causes the form to crash when the input is clicked
+const LabelAndTooltip = styled.div`
+  align-items: center;
+  display: flex;
+  margin-block-end: 0.25rem;
+`;
+
+const Uploader = styled.div`
+  border-radius: 0.1875rem;
+  border: 0.0625rem dashed ${({ theme }) => theme.palette.grey['400']};
+  > * {
+    padding-block: 0.875rem;
+    padding-inline: 1.1rem;
+  }
+`;
+
+const Dropzone = styled.div<{ $isDragActive: boolean; $isDragReject: boolean }>`
+  cursor: pointer;
+  flex-direction: column;
+  inline-size: 100%;
+  text-align: center;
+  transition: background-color 100ms ease;
+
+  :hover,
+  :active,
+  :focus-visible {
+    background-color: #f4f9ff;
+  }
+
+  ${({ $isDragActive, $isDragReject }) => {
+    if ($isDragReject)
+      return css`
+        background-color: #fff6f5;
+      `;
```

Comment:
`isDragReject` doesn’t actually work in a bunch of environments because, for privacy, modern browsers don’t actually disclose any information about the dragged file(s)

I have no reason to believe it wouldn’t work in an old, less stringent browser; but perhaps there’s a good case for to removing this just to clean things up?

Reply:
Decided to remove; it feels tidier